xdsl/transforms/convert_memref_to_ptr.py

```diff
@@ -140,6 +140,53 @@ def match_and_rewrite(self, op: memref.LoadOp, rewriter: PatternRewriter, /):
         rewriter.replace_matched_op(ops, new_results=[load_result.res])
 
 
+@dataclass
+class ConvertSubviewOp(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: memref.SubviewOp, rewriter: PatternRewriter, /):
+        assert isa(memref_type := op.source.type, memref.MemRefType)
+
+        static_offsets = cast(tuple[int, ...], op.static_offsets.get_values())
+        offsets: list[SSAValue] = []
+
+        for idx, offset in enumerate(static_offsets):
+            if offset == memref.SubviewOp.DYNAMIC_INDEX:
+                offsets.append(op.offsets[idx])
+            else:
+                rewriter.insert_op_before_matched_op(
+                    const_op := arith.ConstantOp(
+                        builtin.IntegerAttr(offset, builtin.IndexType())
+                    )
+                )
+                offsets.append(const_op.result)
```

Comment:
nit: I'd rather set names to things if possible
```suggestion
                rewriter.insert_op_before_matched_op(
                    const_op := arith.ConstantOp(
                        builtin.IntegerAttr(offset, builtin.IndexType())
                    )
                )
                const_op.name_hint = f"c{offset.value.data}"
                offsets.append(const_op.result)
```